lib/Conversion/TorchToTcp/TcpCustomOp.cpp

```diff
@@ -117,6 +117,150 @@ class ConvertAtenConvolutionOp : public OpConversionPattern<AtenConvolutionOp> {
+    helper.addOperand("self", adaptor.getSelf());
+    helper.addFloatAttr("scale", op.getScale());
+    helper.addIntAttr("zero_point", op.getZeroPoint());
+    helper.addIntAttr("quant_min", op.getQuantMin());
+    helper.addIntAttr("quant_max", op.getQuantMax());
+
+    return helper.replace();
+  }
+};
+
+class ConvertAtenFakeQuantizePerTensorAffineTensorQparamsOp
+    : public OpConversionPattern<
+          AtenFakeQuantizePerTensorAffineTensorQparamsOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(AtenFakeQuantizePerTensorAffineTensorQparamsOp op,
+                  OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    torch_to_tcp::TorchToTcpCustomOpConversionHelper helper{op, rewriter,
+                                                            getTypeConverter()};
+    helper.addOperand("self", adaptor.getSelf());
+    helper.addIntAttr("quant_min", op.getQuantMin());
+    helper.addIntAttr("quant_max", op.getQuantMax());
+
+    // scale
+    if (!op.getScale().getDefiningOp())
+      return rewriter.notifyMatchFailure(op, "Missing scale operation");
+    auto scaleTensor = dyn_cast<torch::Torch::ValueTensorLiteralOp>(
+        op.getScale().getDefiningOp());
+    if (!scaleTensor)
+      return rewriter.notifyMatchFailure(
+          op, "Scale operation is not ValueTensorLiteralOp");
+    auto scaleElements =
+        dyn_cast<DenseFPElementsAttr>(scaleTensor.getValueAttr());
+    if (!scaleElements || scaleElements.getNumElements() != 1)
+      return rewriter.notifyMatchFailure(op, "Unsupported scale type or size");
+    auto scale = (*scaleElements.begin()).convertToDouble();
+    helper.addDenseFloatArrayAttr("scale", {scale});
+
+    // zero_point
+    if (!op.getZeroPoint().getDefiningOp())
+      return rewriter.notifyMatchFailure(op, "Missing zero point operation");
+    if (dyn_cast<torch::Torch::AtenZerosOp>(
+            op.getZeroPoint().getDefiningOp()) ||
+        dyn_cast<torch::Torch::AtenZerosLikeOp>(
+            op.getZeroPoint().getDefiningOp())) {
+      helper.addDenseIntArrayAttr("zero_point", {0});
+      return helper.replace();
+    }
+    auto zeroPointTensor = dyn_cast<torch::Torch::ValueTensorLiteralOp>(
+        op.getZeroPoint().getDefiningOp());
+    if (!zeroPointTensor)
+      return rewriter.notifyMatchFailure(
+          op,
+          "Zero point operation is not ValueTensorLiteralOp or Zero operation");
+    auto zeroPointElements =
+        dyn_cast<DenseIntElementsAttr>(zeroPointTensor.getValueAttr());
+    if (!zeroPointElements || zeroPointElements.getNumElements() != 1)
+      return rewriter.notifyMatchFailure(op,
+                                         "Unsupported zero point type or size");
+    auto zeroPoint = (*zeroPointElements.begin()).getSExtValue();
+    helper.addDenseIntArrayAttr("zero_point", {zeroPoint});
```

Comment:
Consider rewriting this section as follows to make control flow more readable and less error prone to future modifications:
```C++
auto zeroPointOp = op.getZeroPoint().getDefiningOp();

if (dyn_cast<torch::Torch::AtenZerosOp>(zeroPointOp) ||
  dyn_cast<torch::Torch::AtenZerosLikeOp>(zeroPointOp)) {
  auto zeroPoint = -=0;
} else {
  auto zeroPointTensor = ...
  ...
  auto zeroPoint = (*zeroPointElements.begin()).getSExtValue();
}
helper.addDenseIntArrayAttr("zero_point", {zeroPoint});
return helper.replace();
```